cns/restserver/restserver.go

```diff
@@ -1252,7 +1263,31 @@ func (service *HTTPRestService) getNetworkContainerResponse(req cns.GetNetworkCo
 		}
 
 		logger.Printf("pod info %+v", podInfo)
-		containerID = service.state.ContainerIDByOrchestratorContext[podInfo.PodName+podInfo.PodNamespace]
+
+		var (
+			context      = podInfo.PodName + podInfo.PodNamespace
+			dncEP        = service.GetOption(acn.OptPrivateEndpoint).(string)
+			infraVnet    = service.GetOption(acn.OptInfrastructureNetwork).(string)
+			nodeID       = service.GetOption(acn.OptNodeID).(string)
+			isManagedDnc = dncEP != "" && infraVnet != "" && nodeID != ""
+		)
+
+		containerID, exists = service.state.ContainerIDByOrchestratorContext[context]
+		if !exists {
+			if isManagedDnc {
+				service.lock.Unlock()
```

Comment:
why do you unlock?..no lock acquired before this code

Reply:
it's actually supposed to be RUnlock and RLock